mindsdb/interfaces/database/database.py

```diff
@@ -48,17 +49,24 @@ def _get_integrations(self):
 
     def register_predictors(self, model_data_arr, setup=True):
         it = self._get_integrations()
-        for integration in it:
-            register = True
-            if setup:
-                register = self._setup_integration(integration)
-            if register:
-                if integration.check_connection():
-                    integration.register_predictors(model_data_arr)
-                else:
-                    logger.warning(f"There is no connection to {integration.name}. predictor wouldn't be registred.")
+        for model in [*model_data_arr,None]:
+            if model is None:
+                model_arr = []
+            else:
+                model_arr = [model]
+            try:
+                for integration in it:
+                    register = True
+                    if setup:
+                        register = self._setup_integration(integration)
+                    if register:
+                        if integration.check_connection():
+                            integration.register_predictors(model_arr)
```

Comment:
after predictor training finished, that line called 2 times for same integration. That leads to this output:
```
--- Logging error ---
Traceback (most recent call last):
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/interfaces/database/database.py", line 65, in register_predictors
    integration.register_predictors(model_arr)
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/integrations/postgres/postgres.py", line 140, in register_predictors
    self._query(q)
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/integrations/postgres/postgres.py", line 52, in _query
    cur.execute(query)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/pg8000/core.py", line 865, in execute
    self._c.execute(self, operation, args)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/pg8000/core.py", line 1925, in execute
    self.handle_messages(cursor)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/pg8000/core.py", line 1992, in handle_messages
    raise self.error
pg8000.core.ProgrammingError: {'S': 'ERROR', 'V': 'ERROR', 'C': '42P07', 'M': 'relation "carp_p4" already exists', 'F': 'heap.c', 'L': '1155', 'R': 'heap_create_with_catalog'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/logging/__init__.py", line 994, in emit
    msg = self.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 840, in format
    return fmt.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 577, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.6/logging/__init__.py", line 338, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.6/multiprocessing/spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "/usr/lib/python3.6/multiprocessing/spawn.py", line 118, in _main
    return self._bootstrap()
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/interfaces/native/predictor_process.py", line 45, in run
    state.update_predictor(name=name, status=status, original_path=None, data=json.dumps(analysis))
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/interfaces/state/state.py", line 73, in update_predictor
    }], False)
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/interfaces/database/database.py", line 70, in register_predictors
    logger.error('Cloud not regier model: ', e)
Message: 'Cloud not regier model: '
Arguments: (ProgrammingError({'S': 'ERROR', 'V': 'ERROR', 'C': '42P07', 'M': 'relation "carp_p4" already exists', 'F': 'heap.c', 'L': '1155', 'R': 'heap_create_with_catalog'},),)
```

Reply:
Same for me